manifests/database.pp

```diff
@@ -129,6 +131,7 @@
   $database_host_ip                = $zabbix::params::server_database_host_ip,
   $database_charset                = $zabbix::params::server_database_charset,
   $database_collate                = $zabbix::params::server_database_collate,
+  $database_tablespace             = $zabbix::params::server_database_tablespace,
```

Comment:
Can you please add a datatype here? Probably a string:
```suggestion
  Optional[String[1]] $database_tablespace = $zabbix::params::server_database_tablespace,
```

Reply:
I fail to see why: the parameter always gets a default value from the params class.

Reply:
Because we want database for all classes which aren't private :)

Reply:
?

Reply:
I think that should have read...

> We want data *types* for all classes which aren't private.

New parameters should be given types, even if there are other parameters that don't yet have any defined.

Reply:
I can see how `String` would fit, but not `Optional`, since this parameter always has a value (`$zabbix::params::server_database_tablespace`) and will never cause a failure when the parameter is omitted.

Reply:
> I think that should have read...
> 
> > We want data _types_ for all classes which aren't private.
> 
> New parameters should be given types, even if there are other parameters that don't yet have any defined.

👍